### PR TITLE
Ensure slf4j message diagnostic context cleanup

### PR DESCRIPTION
**Pull request by schlosna**

### CODE REVIEW

**Comment by schlosna** [comment]
@uschi2000 minor fix in prep for more tracing

File: http-servers/src/main/java/com/palantir/remoting1/servers/TraceIdLoggingFilter.java [hunk 1]
```diff
@@ -39,16 +40,32 @@ public void init(FilterConfig filterConfig) throws ServletException {}
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
             throws IOException, ServletException {
+        try (Closeable ignored = populateTraceContext(request)) {
```
Comment (uschi2000) [inline http-servers/src/main/java/com/palantir/remoting1/servers/TraceIdLoggingFilter.java]: but it's not ignored, right?

Comment (uschi2000) [inline http-servers/src/main/java/com/palantir/remoting1/servers/TraceIdLoggingFilter.java]: (in the sense that the try-with-resource construct will clean up the MDC)

Comment (uschi2000) [inline http-servers/src/main/java/com/palantir/remoting1/servers/TraceIdLoggingFilter.java]: maybe add a comment here explaining how this construction works.

Comment (schlosna) [inline http-servers/src/main/java/com/palantir/remoting1/servers/TraceIdLoggingFilter.java]: Went with more explicit try/finally. 

Was previously `ignored` as Idea doesn't warn on unused var in try-with-resources (though it is implicitly used & closed).

**Pull request merged**